tests/helpers/providerStateHelper.js

```diff
@@ -50,25 +50,87 @@ const givenFolderExists = (provider, username, password, resource) => {
 }
 
 /**
- * provider state: creates a share
+ * provider state: creates a user share
  *
  * @param {object} provider
  * @param {string} username
- * @param {sring} password
- * @param {sring} resource
- * @param {sring} shareType
- * @param {sring} shareWith
+ * @param {sring} userPassword
+ * @param {sring} path
+ * @param {string} shareWith
+ * @param {object} optionalParams
  */
-const givenShareExists = async (
+const givenUserShareExists = (
   provider,
   username,
-  password,
-  resource,
-  shareType,
-  shareWith
+  userPassword,
+  path,
+  shareWith,
+  optionalParams
+) => {
+  return provider
+    .given('resource is shared', {
+      username,
+      userPassword,
+      path,
+      shareType: 0,
+      shareWith,
+      ...optionalParams
+    })
+}
+
+/**
+ * provider state: creates a group share
+ *
+ * @param {object} provider
+ * @param {string} username
+ * @param {sring} userPassword
+ * @param {sring} path
+ * @param {string} shareWith
+ * @param {object} optionalParams
+ */
+const givenGroupShareExists = (
+  provider,
+  username,
+  userPassword,
+  path,
+  shareWith,
+  optionalParams
+) => {
+  return provider
+    .given('resource is shared', {
+      username,
+      userPassword,
+      path,
+      shareType: 1,
```

Comment:
```suggestion
      shareType: SHARE_TYPE.<some-name>,
```
please assign these numbers with some vars. it would be a lot easier to understand what these magic numbers mean